packages/teleport/src/lib/tdp/codec.ts

```diff
@@ -328,26 +353,34 @@ export default class Codec {
     return message;
   }
 
-  // decodeRegion decodes the region from a PNG_FRAME tdp message.
-  decodeRegion(buffer: ArrayBuffer): Region {
+  // decodePngFrame decodes a raw tdp PNG frame message and returns it as a PngFrame
+  // | message type (2) | left uint32 | top uint32 | right uint32 | bottom uint32 | data []byte |
+  // https://github.com/gravitational/teleport/blob/master/rfd/0037-desktop-access-protocol.md#2---png-frame
+  decodePngFrame(buffer: ArrayBuffer, onload: (PngFrame) => any): PngFrame {
```

Comment:
Refactored in order to create a pattern of having `decodeTdpMessage(buffer: ArrayBuffer): TdpMessage` (another example is `decodeClientScreenSpec(buffer: ArrayBuffer): ClientScreenSpec`.

However in this particular case, I needed to pass an `onload` function in since the `onload` must be set before `data.src`, otherwise there's a race condition (see [here](https://github.com/gravitational/webapps/pull/572/files#diff-f7aac69b83399d0d34d5e42582b6a8e1c04375ef9205491f486f417191a93b9bR136-R139) for how it's used).